### PR TITLE
sink(ticdc): add table sink advancer

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8132

### What is changed and how it works?

- add table sink advancer and test it
- refine table sink worker test
#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Test coverage:
image

File: cdc/processor/sinkmanager/table_sink_advancer.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+}
+
+// tryAdvanceAndAcquireMem tries to acquire the memory quota and advance the table sink.
+// allFinished indicates whether all the events have been fetched. Then we
+// do not need to acquire the memory quota anymore.
+// txnFinished indicates whether the current transaction has been finished.
+// If it is finished, it is OK to wait next round task to advance the table sink.
+// Otherwise, we need to advance the table at least to the current transaction.
+func (a *tableSinkAdvancer) tryAdvanceAndAcquireMem(
+	allFinished bool,
```
Comment (amyangfei) [inline cdc/processor/sinkmanager/table_sink_advancer.go:167]: Is `allFetched` more accurate according to comments?

File: cdc/processor/sinkmanager/table_sink_worker.go [hunk 2]
```diff
@@ -108,237 +108,25 @@ func validateAndAdjustBound(changefeedID model.ChangeFeedID,
...
-		}
-	}
-	return emitted, nil
-}
-
 func (w *sinkWorker) handleTask(ctx context.Context, task *sinkTask) (finalErr error) {
-	lowerBound, upperBound := validateAndAdjustBound(w.changefeedID, task)
+	advancer := newTableSinkAdvancer(task, w.splitTxn, w.sinkMemQuota, requestMemSize)
+	// The task is finished and some required memory isn't used.
+	defer advancer.cleanup()
```
Comment (hicqu) [inline cdc/processor/sinkmanager/table_sink_worker.go:114]: Considering this situation, with splitTxn = true:
1. handles part of a large transaction, commitTs  = 1 and batchID = 1;
2. the task is returned to sinkProgress heap;
3. the table is rescheduled, and advanced to commitTs = 1 and batchID = 1 again.

I think it's possble. To resolve this problem, we can store Advancers in sinkProgress or tableSinkWrapper directly.

Comment (hicqu) [reply cdc/processor/sinkmanager/table_sink_worker.go:114]: Or we can make `batchID` as an global atomic variable. I guess it's a better fix in release-6.5.

Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_worker.go:114]: Good catch! I'll fix this bug later! Thanks!

**Review by hicqu — APPROVED** [review]

File: cdc/processor/sinkmanager/table_sink_advancer.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_advancer.go]: Updated! Thanks!

**Review by amyangfei — APPROVED** [review]

**Pull request merged**